diff -Nru xf86-video-apm-1.2.0/src/apm.h xf86-video-apm-1.2.0-shd/src/apm.h
--- xf86-video-apm-1.2.0/src/apm.h	2008-03-19 15:05:14.000000000 +0000
+++ xf86-video-apm-1.2.0-shd/src/apm.h	2008-10-06 13:58:59.324489000 +0000
@@ -243,7 +243,7 @@
 extern void	ApmAccelReserveSpace(ApmPtr pApm);
 
 extern int	ApmPixmapIndex;
-#define APM_GET_PIXMAP_PRIVATE(pix)\
-	((ApmPixmapPtr)(((PixmapPtr)(pix))->devPrivates[ApmPixmapIndex].ptr))
+//#define APM_GET_PIXMAP_PRIVATE(pix)	((ApmPixmapPtr)(((PixmapPtr)(pix))->devPrivates[ApmPixmapIndex].ptr))
+#define APM_GET_PIXMAP_PRIVATE(pix)	((ApmPixmapPtr)(((PixmapPtr)(pix))->devPrivates))
 
 #include "apm_regs.h"
diff -Nru xf86-video-apm-1.2.0/src/apm_rush.c xf86-video-apm-1.2.0-shd/src/apm_rush.c
--- xf86-video-apm-1.2.0/src/apm_rush.c	2008-03-19 15:05:14.000000000 +0000
+++ xf86-video-apm-1.2.0-shd/src/apm_rush.c	2008-10-06 13:48:32.365306500 +0000
@@ -8,6 +8,8 @@
 
 #include "apm.h"
 #include "xaalocal.h"
+#include "dix.h"
+#include "privates.h"
 
 extern DriverRec APM;
 
@@ -544,7 +546,7 @@
 
     REQUEST_SIZE_MATCH(xXF86RushOverlayPixmapReq);
 
-    VALIDATE_DRAWABLE_AND_GC(stuff->drawable, pDraw, pGC, client);
+    VALIDATE_DRAWABLE_AND_GC(stuff->drawable, pDraw, pGC);
 
     pScrn = xf86Screens[pDraw->pScreen->myNum];
     pApm = APMPTR(pScrn);
